src/modules/lighthouse/Eth2Client.ts

```diff
@@ -304,17 +320,19 @@ export class Eth2Client {
     const retry = retrier(this.logger, maxRetries, 100, 10000, true);
     const request = async () => {
       const res = <{ dependent_root: string, data: ProposerDutyInfo[] }>(
-        await this.retryGet(this.endpoints.proposerDutes(epoch), 5, this.apiGetNoFallback, false).catch(
+        await this.retryRequest(
+          (rpcURL: string) => this.apiGet(rpcURL, this.endpoints.proposerDutes(epoch)),
+          {maxRetries, dataOnly: false}
+        ).catch(
           (e) => {
             this.logger.error('Unexpected status code while fetching proposer duties info');
             throw e;
           }
         )
       );
       if (res.dependent_root != dependentRoot) {
-        throw new BeaconChainGeneralApiError(errCommon(
+        throw Error(
```

Comment:
Why you switch to general error here?

Reply:
`BeaconChainGeneralApiError` should only be used in basic requests functions